pkg/deploy/util/util.go

```diff
@@ -77,6 +78,10 @@ func SetDeploymentCondition(status *deployapi.DeploymentConfigStatus, condition
 	if currentCond != nil && currentCond.Status == condition.Status && currentCond.Reason == condition.Reason {
 		return
 	}
+	// Preserve lastTransitionTime if we are not switching between statuses of a condition.
+	if currentCond != nil && currentCond.Status == condition.Status {
```

Comment:
We definitely need this for 1.4.  I don't know that we need the rest.  Can you split this out?